### PR TITLE
ISSITES-5787 Top Level Public Sites Nav changes

### PR DESCRIPTION
**Pull request by liaoethan**

### CODE REVIEW

**Comment by jotchon** [comment]
Just started reviewing :)

File: modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/fragments/group/public-sites-navigation/fragments/public-sites-navigation/index.scss [hunk 1]
```diff
@@ -39,26 +39,28 @@ nav.public-sites-navigation {
...
+
+							.adt-nav-caret-bottom-icon {
+								color: var(--white);
+							}
 						}
 
-						.adt-nav-caret-bottom-icon {
-							color: var(--white);
+						.adt-nav-text {
+							.adt-nav-title {
```
Comment (jotchon) [inline modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/fragments/group/public-sites-navigation/fragments/public-sites-navigation/index.scss]: If there are no styles on the parent CSS selector (i.e. .adt-nav-text), ideally we would just keep it on the same line like .adt-nav-text .adt-nav-title instead of nesting.

File: modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/ddm-templates/navigation-utilities/ddm-template.ftl [hunk 2]
```diff
@@ -64,14 +64,10 @@
 											<div class="adt-submenu-item-preheader color-neutral-3 font-weight-semi-bold">${preheaderText}</div>
 										</#if>
 
-										<#if stringUtil.equals(menuItemType, "Image")>
```
Comment (jotchon) [inline modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/ddm-templates/navigation-utilities/ddm-template.ftl]: Are we removing the ability to add images to the navigation?

File: modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/style-books/learn/frontend-tokens-values.json [hunk 3]
```diff
@@ -42,6 +42,10 @@
 		"cssVariableMapping": "color-brand-primary-darken-5",
 		"value": "#00256C"
 	},
+	"colorNeutral2": {
+		"cssVariableMapping": "color-neutral-2",
```
Comment (jotchon) [inline modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/style-books/learn/frontend-tokens-values.json:50]: You could squash the two commits that add color to frontend tokens.  No need for them to be separate commits.

File: modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/fragments/group/public-sites-navigation/fragments/public-sites-navigation/index.scss [hunk 4]
```diff
@@ -389,53 +402,79 @@ html body:not(.has-edit-mode-menu) .content #main-content.layout-content {
...
 	.adt-nav-text:focus {
 		outline: none;
 	}
 
 	&.dropdown-open {
-		.adt-angle-down-svg {
-			transform: translate3d(0px, 0px, 0px) scale3d(1, 1, 1)
-				rotateX(180deg) rotateY(0deg) rotateZ(0deg)
-				skew(0deg, 0deg);
+		.adt-nav-caret-bottom-icon {
```
Comment (jotchon) [inline modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/fragments/group/public-sites-navigation/fragments/public-sites-navigation/index.scss]: Is the rotation not possible anymore?

File: modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/ddm-templates/navigation-utilities/ddm-template.ftl [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
 						${navPrimaryItem.getName()}
-
-						<span class="adt-angle-down-svg">
-							<svg aria-hidden="true" class="lexicon-icon lexicon-icon-angle-down" focusable="false"><use href="/o/osb-www-theme/images/clay/icons.svg#angle-down"></use></svg>
-						</span>
+					</span>
+					<span class="adt-nav-caret-bottom-icon align-self-center">
+						<svg class="lexicon-icon lexicon-icon-caret-bottom" role="presentation" viewBox="0 0 512 512"><use xlink:href="/o/admin-theme/images/clay/icons.svg#caret-bottom"></use></svg>
+					</span>
+					<span class="adt-nav-caret-top-icon align-self-center">
```
Comment (jotchon) [inline modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/ddm-templates/navigation-utilities/ddm-template.ftl]: Is there a reason for adding more markup?  Previously the caret was rotating.

**Review by jotchon — COMMENTED** [review]
It looks like the Public Site Nav is coming along well!  There were a few things that were not working when I was testing:
- SiteNavigationMenuItem configurations are not working, specifically the dropdowns
    - Dropdown don’t have any selections
    - Image custom field is missing
- Up/Down Caret:  Could we not achieve this like we did previously with the rotation?
- Menu does not open, should this be working?
- Nav dropdowns should be right aligned, but they are currently centered

This is not apart of the ticket, but Abel mentioned awhile ago that we need to change the logo.  It should just be logo and subsite name, not logo, Liferay, and subsite name.  Do you mind adding that to this ticket?  It should be simple.
Screen Shot 2022-11-02 at 5 36 58 AM

Let me know if you have any questions about my comments above.

**Comment by jotchon** [comment]
More notes:
- The box shadow is also missing from the mega dropdown menu.
- Note to use "dropdown-open" since the dropdown provider is not available yet
- Expando Value Permissions need to be viewable for guests

File: modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/ddm-templates/navigation-utilities/ddm-template.ftl (hunk 2 above)
Comment (liaoethan) [reply modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/ddm-templates/navigation-utilities/ddm-template.ftl]: For this section that was removed, it was for the heading change if the menu item type was an image
```
<#if stringUtil.equals(menuItemType, "Image")>
	<div class="adt-submenu-item-title font-size-paragraph-small font-weight-semi-bold">${navTertiaryItem.getName()}</div>
<#else>
	<div class="adt-submenu-item-title font-size-paragraph-base font-weight-semi-bold">${navTertiaryItem.getName()}</div>
</#if>
```

This doesn't apply to the new navigation. The images are still there in L58-60:
```
	<#if stringUtil.equals(menuItemType, "Image") && imageURL?has_content>
		<img class="adt-submenu-item-image" loading="lazy" src="${imageURL}" />
	</#if>
```

Comment (liaoethan) [reply modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/ddm-templates/navigation-utilities/ddm-template.ftl]: But this doesn't apply to the new navigation

**Comment by liaoethan** [comment]
Hi @jotchon, I made the following changes:

- Squashed redundant commits (stylebook tokens, etc.)
- Removed the top caret
- Make refactoring changes with scss
- Added the box shadow
- Right aligned menu items
- Removal of liferay text from liferay logo

For the image configuration change, I added a comment. Let me know if you want me to make any additional changes. Thanks for all of the feedback!

**Comment by jotchon** [comment]
Just started reviewing :)

**Comment by jotchon** [comment]
Hi @liaoethan, there are few things that could be improved with this PR:

- it could be more organized.  There are a lot of commits and it is difficult to review.
- Many of the commits say "refactor".  This could be more descriptive.
- The squashed commits have two lined messages (i.e. color-neutral commit).  I believe you forgot to comment out the other commit message.
- since this is the first commit for the ADT, you could combine those changes into one commit.  You might keep those changes separate for yourself, but we don't need to see all the changes for this.  

Not sure if this is clear, but let me know if you have questions about this.  Otherwise, this is great work!

**Comment by liaoethan** [comment]
Hi @jotchon, thanks for the feedback. I'll create another PR and make the commits more concise.

**Comment by liaoethan** [comment]
I created a condensed pull here: https://github.com/jotchon/liferay-portal/pull/4

I see extra commits but is that being caused by an outdated master?
